Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -1166,10 +1177,44 @@ WarpXOpenPMDPlot::GetMeshCompNames (int meshLevel,
     }
 
     if ( 0 == meshLevel )
-      return;
+        return;
 
     field_name += std::string("_lvl").append(std::to_string(meshLevel));
 }
+/* Split varname into name and theta mode mode index
+ *
+ * @param varname [IN]: variable name
+ * @param varname field_str_end_index [OUT] : if varname = fieldName_modeNumber_realOrImag,  index of underscore after fieldName, otherwise, varname.size()
+ * @param varname mode_index [OUT] : if varname = fieldName_modeNumber_realOrImag, returns 2 * modeNumber - 1 + (realOrImag == 'imag'), otherwise, -1
```

Comment:
Doxygen syntax is this:
```suggestion
 * @param[in] varname variable name
 * @returns varname field_str_end_index if varname = fieldName_modeNumber_realOrImag,  index of underscore after fieldName, otherwise, varname.size() and if varname = fieldName_modeNumber_realOrImag, returns 2 * modeNumber - 1 + (realOrImag == 'imag'), otherwise, -1
```

Maybe `varname variable name` you can find a more descriptive string? :)

Reply:
Potentially, we can implement most of the parsing logic here with a `<regex>`.

I use https://regex101.com to build complex expressions.

Reply:
Ok, maybe the regex would be easier to read and less cryptic than my use of string lengths.  I thought using certain positions in the strings might run faster than string searches.
Do you have a preference of 1. checking positions in strings  vs 2. string compare, contains functions  vs 3. regex expressions?

Reply:
The other advantage to the current approach is a robust way of extracting the field name.  It is important to split
rho_species_1_23_real into  rho_species_1  , pull out the 23, and identify the real

Reply:
I like the regex version - this is now in the PR